src/EFCore.PG/ValueGeneration/Internal/NpgsqlValueGeneratorSelector.cs

```diff
@@ -104,6 +104,6 @@ public override bool TrySelect(IProperty property, ITypeBase typeBase, out Value
         => property.ClrType.UnwrapNullableType() == typeof(Guid)
             ? property.ValueGenerated == ValueGenerated.Never || property.GetDefaultValueSql() is not null
                 ? new TemporaryGuidValueGenerator()
-                : new GuidValueGenerator()
+                : new UUid7ValueGenerator()
```

Comment:
We should have some tests on this line as well as on the generator's implementation. I think we should at least include:

- UUIDv7 should be generated by default.
- The version and variant bits should be set as expected.
- The timestamp bits should reflect the timestamp (at least coarsely asserted).
- The final bits should have a high entropy.

Reply:
IMHO there's a bit of diminishing returns on this kind of testing - the test basically ends up duplicating the implementation. I do agree it makes sense to make sure that the generator is actually being used by default (e.g. that the version/variant bits are correct), but beyond that I personally wouldn't invest too more effort in e.g. asserting on the timestamp.

Reply:
Not to mention that this is all supposed to go away in the next release...

Reply:
I already have a test in EFCore.Jet that I can copy to here. Just need to know where to put it

Reply:
Fair. I did not mean to imply that we test the UUIDv7 implementation's inner workings, but mainly assert the fact _that_ UUIDv7 is being used. I agree that the version and variant are the primary indicators.